lib/credentials/shared_ini_file_credentials.js

```diff
@@ -49,19 +49,38 @@ AWS.SharedIniFileCredentials = AWS.util.inherit(AWS.Credentials, {
    *   role profile is selected, an error is raised.
    * @option options preferStaticCredentials [Boolean] (false) True to
    *   prefer static credentials to role_arn if both are present.
+   * @option options tokenCodeFn [Function] (noop function) Function to provide
+   *   STS Assume Role TokenCode, if mfa_serial is provided for profile in ini
+   *   file. Function is called with value of mfa_serial and callback, and
+   *   should provide the TokenCode or an error to the callback in the format
+   *   callback(err, token)
    */
   constructor: function SharedIniFileCredentials(options) {
     AWS.Credentials.call(this);
 
     options = options || {};
 
+    this.refreshCallbacks = [];
     this.filename = options.filename;
     this.profile = options.profile || process.env.AWS_PROFILE || AWS.util.defaultProfile;
     this.disableAssumeRole = Boolean(options.disableAssumeRole);
     this.preferStaticCredentials = Boolean(options.preferStaticCredentials);
+    this.tokenCodeFn = options.tokenCodeFn || function(mfaSerial, callback) {
```

Comment:
Instead of defining a default function if one isn't provided, can you update the `if (mfaSerial)` check in `loadRoleProfile` to also check for the existence of `this.tokenCodeFn`? That will save us from having to create an extra no-op function.